src/m365/spo/commands/site/site-remove.ts

```diff
@@ -76,259 +80,185 @@ class SpoSiteRemoveCommand extends SpoCommand {
-            }
+        if (!siteDetails.TimeDeleted) {
+          throw `Site '${options.url}' is not in the recycle bin. Remove it first.`;
+        }
 
-            await this.deleteGroup(group.id, logger);
-            await this.deleteSite(args.options.url, args.options.wait, logger);
+        if (isGroupSite) {
+          const isGroupPermanentlyDeleted = await this.isGroupPermanentlyDeleted(logger, siteDetails.GroupId);
+          if (!isGroupPermanentlyDeleted) {
+            await this.removeGroupFromEntraRecycleBin(logger, siteDetails.GroupId);
           }
-          catch (err: any) {
-            if (this.verbose) {
-              await logger.logToStderr(`Site group doesn't exist. Searching in the Microsoft 365 deleted groups.`);
-            }
-
-            const deletedGroups = await this.isSiteGroupDeleted(groupId);
-            if (deletedGroups.value.length === 0) {
-              if (this.verbose) {
-                await logger.logToStderr("Site group doesn't exist anymore. Deleting the site.");
-              }
-
-              if (args.options.wait) {
-                await logger.logToStderr(chalk.yellow(`Entered site is a groupified site. Hence, the parameter 'wait' will not be applicable.`));
-              }
-
-              await this.deleteOrphanedSite(logger, args.options.url);
-            }
-            else {
-              throw `Site group still exists in the deleted groups. The site won't be removed.`;
-            }
+        }
+        await this.deleteSiteFromSharePointRecycleBin(logger, options.url);
+      }
+      else {
+        if (siteDetails.TimeDeleted) {
+          throw `Site '${options.url}' is already in the recycle bin. Use --fromRecycleBin to permanently delete it.`;
+        }
+
+        if (isGroupSite) {
+          await this.deleteGroupifiedSite(logger, options.url);
+          if (options.skipRecycleBin) {
+            await this.removeGroupFromEntraRecycleBin(logger, siteDetails.GroupId);
           }
         }
+        else {
+          await this.deleteNonGroupSite(logger, options.url);
+        }
+
+        if (options.skipRecycleBin) {
+          await this.deleteSiteFromSharePointRecycleBin(logger, options.url);
+        }
       }
     }
     catch (err: any) {
       this.handleRejectedPromise(err);
     }
   }
 
-  private isSiteGroupDeleted(groupId: string): Promise<{ value: { id: string }[] }> {
-    const requestOptions: any = {
-      url: `https://graph.microsoft.com/v1.0/directory/deletedItems/Microsoft.Graph.Group?$select=id&$filter=groupTypes/any(c:c+eq+'Unified') and startswith(id, '${groupId}')`,
-      headers: {
-        accept: 'application/json;odata.metadata=none'
-      },
-      responseType: 'json'
-    };
-
-    return request.get<{ value: { id: string }[] }>(requestOptions);
-  }
+  private async removeGroupFromEntraRecycleBin(logger: Logger, groupId: string): Promise<void> {
```

Comment:
Noticed that when I try to remove a site from the recycle bin, I'm getting an error. Seems like it takes a couple of seconds before the group is actually present in the recycle bin.
We should probably poll for the group in the recycle bin before deleting it. Maybe we can use the function `isGroupInRecycleBin` for this.